dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/contig/ContigMappingTest.java

```diff
@@ -0,0 +1,92 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.dbsnpimporter.contig;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+
+public class ContigMappingTest {
+
+    private static final String GENBANK_1 = "genbank_example_1";
+
+    private static final String GENBANK_2 = "genbank_example_2";
+
+    private static final String GENBANK_3 = "genbank_example_3";
+
+    private static final String REFSEQ_1 = "refseq_example_1";
+
+    private static final String REFSEQ_2 = "refseq_example_2";
+
+    private static final String REFSEQ_3 = "refseq_example_3";
+
+    private static final String GENBANK_CONTIG = "GL456213.1";
+
+    private static final String REFSEQ_CONTIG = "NT_166283.1";
+
+    private static final String REFSEQ_CONTIG_WITHOUT_SYNONYM = "NT_without_synonym";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void useMapContructor() throws Exception {
+        HashMap<String, String> contigMap = new HashMap<>();
+        contigMap.put(REFSEQ_1, GENBANK_1);
+        contigMap.put(REFSEQ_2, GENBANK_2);
+        contigMap.put(REFSEQ_3, GENBANK_3);
+        ContigMapping contigMapping = new ContigMapping(contigMap);
+
+        assertEquals(GENBANK_1, contigMapping.getGenbankOrDefault(REFSEQ_1));
+        assertEquals(GENBANK_2, contigMapping.getGenbankOrDefault(REFSEQ_2));
+        assertEquals(GENBANK_3, contigMapping.getGenbankOrDefault(REFSEQ_3));
+    }
+
+    @Test
+    public void noAvailableMapping() throws Exception {
+        HashMap<String, String> contigMap = new HashMap<>();
+        contigMap.put(REFSEQ_1, GENBANK_1);
+        ContigMapping contigMapping = new ContigMapping(contigMap);
+
+        assertEquals(REFSEQ_CONTIG_WITHOUT_SYNONYM, contigMapping.getGenbankOrDefault(REFSEQ_CONTIG_WITHOUT_SYNONYM));
+    }
+
+    @Test
+    public void useFileConstructor() throws Exception {
+        String assemblyReport = "AssemblyReport.txt";
+        String fakeFtpLocation = Thread.currentThread().getContextClassLoader().getResource(assemblyReport).toString();
+
+        ContigMapping contigMapping = new ContigMapping(fakeFtpLocation);
+
+        assertEquals(GENBANK_CONTIG, contigMapping.getGenbankOrDefault(REFSEQ_CONTIG));
+        assertEquals(REFSEQ_CONTIG_WITHOUT_SYNONYM, contigMapping.getGenbankOrDefault(REFSEQ_CONTIG_WITHOUT_SYNONYM));
+    }
+
+    @Test
+    @Ignore("This test does an external ftp request, which is too slow to be a comfortable test.")
```

Comment:
Is there a way to make Travis to execute this test?

Reply:
remove the ignore :sweat_smile: I disabled it because it can take up to half a minute waiting for the ftp. If we are ok with that both in local test runs and travis, we should enable it. right now we can run it manually without changing code.

Reply:
I prefer not running those slow tests in local. I meant something like add a @Category to the slow test, and exclude or include the test in that category from the maven execution (ideally, exclude them by default, and include them in the travis build with an special mvn command)

https://stackoverflow.com/questions/14132174/how-to-exclude-all-junit4-tests-with-a-given-category-using-maven-surefire